src/map/script.cpp

```diff
@@ -8782,6 +8782,137 @@ BUILDIN_FUNC(getequipuniqueid)
+{
+	int i, k, s = 0;
+	TBL_PC* sd;
+	struct item* it = NULL;
+
+	unsigned long long uniqueid = atoll(script_getstr(st, 2));
+	struct script_data* data = script_getdata(st, 3);
+	struct reg_db* ref = NULL;
+	const char* name;
+	int32 idx, id;
+
+	if (!uniqueid) {
+		ShowError("buildin_delitem_uniqueid: unknown unique id .\n");
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	if (!script_charid2sd(4, sd)) {
+		script_pushint(st, false);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	for (i = 0; i < MAX_INVENTORY; i++) {
+		if (sd->inventory.u.items_inventory[i].unique_id == uniqueid)
+			break;
+	}
+
+	if (!data_isreference(data))
+	{
+		ShowError("buildin_getitemreferences: not a variable\n");
+		script_reportdata(data);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+	name = reference_getname(data);
+	id = reference_getid(data);
+	idx = reference_getindex(data);
+	ref = reference_getref(data);
+
+	if (not_server_variable(*name) && !sd) {
+		ShowError("buildin_getitemreferences: Cannot use a player variable '%s' if no player is attached.\n", name);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+	if (is_string_variable(name)) {
+		ShowError("buildin_getitemreferences: Cannot Assign String variable '%s'.\n", name);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+	it = &sd->inventory.u.items_inventory[i];
+	if (it != NULL) {
+
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->nameid), ref);
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->refine), ref);
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->identify), ref);
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->attribute), ref);
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->card[0]), ref);
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->card[1]), ref);
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->card[2]), ref);
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->card[3]), ref);
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->expire_time), ref);
+		set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->bound), ref);
+
+		for (k = 0; k < MAX_ITEM_RDM_OPT; k++)
+		{
+			set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->option[k].id), ref);
+			set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->option[k].value), ref);
+			set_reg(st, sd, reference_uid(id, idx + s++), name, (void*)__64BPRTSIZE(it->option[k].param), ref);
+		}
+		script_pushint(st, true);
+	}
+	else
+	{
+		ShowError("buildin_getitemreferences: Item not found.\n");
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	return SCRIPT_CMD_SUCCESS;
+}
+/*==========================================
+ * delitem_uniqueid(<"string item unique id">{,<char_id>})
```

Comment:
on success and on failure (return) what the function will return?
In other words add 0 if failed command, > 0 if success or like that I dunno.

Reply:
the problem is , there is 2 options
`st->state = END;` which is noob friendly
or `script_pushint(st, false);` which is better for pros
(unless there is third option that i don't know about)

pros can easily make sure that they don't get an error of open a bug in the script
while noobs might open so many bugs in there scripts without known

ofc everyone should read the doc in the end , but they might miss understand or something
i would rather something like
```cs
if(!uniqueid_delitem(.@unique_id$)){
	logmes "uniqueid_delitem(x) something went wrong, (more information) 'possible exploit'.";
	end;
}
(or doing something before this happen)
```
but most of the users will do
```cs
uniqueid_delitem(.@unique_id$);
getitem3 xxx;
```